App.tsx

```diff
@@ -17,57 +17,82 @@ interface Product {
   discountedPrice: number;
   description: string;
 }
- 
+
 function App(): JSX.Element {
-const [page , setPage] = useState(1);
-const [products, setProducts] = useState<Product[]>([]);
-const [loading, setLoading] = useState(true);
+  const [page, setPage] = useState(1);
+  const [products, setProducts] = useState<Product[]>([]);
+  const [loading, setLoading] = useState(true);
 
-const apiData = async()=>{
-  try{
-    const res = await fetch(`${'https://dummyjson.com/products'}?limit=05&page=${page}`)
-    const data = await res.json();
-    setProducts((prevProducts) => [...prevProducts, ...data?.products]);
-  } catch (error) {
-    console.error('Error fetching products:', error);
-  } finally {
-    setLoading(false);
-  }
-  }
+  const apiData = async () => {
+    try {
+      const res = await fetch(
+        `${'https://dummyjson.com/products'}?limit=05&page=${page}`,
+      );
+      const data = await res.json();
+      setProducts(prevProducts => [...prevProducts, ...data?.products]);
```

Comment:
For safety write like : setProducts(prevProducts => [...prevProducts, ...(data?.products || [])); because data?.products can be undefined or null and applying spread operator will result is error

Reply:
Done